components/system-broker/tests/catalog_test/catalog_test.go

```diff
@@ -0,0 +1,351 @@
+    {
+      "message": "Internal Server Error",
+      "path": [
+        "result"
+      ],
+      "extensions": {
+        "error": "InternalError",
+        "error_code": 10
+      }
+    }
+  ],
+  "data": null
+}`
+)
+
+func TestOSBCatalog(t *testing.T) {
+	suite.Run(t, new(OSBCatalogTestSuite))
+}
+
+type OSBCatalogTestSuite struct {
+	suite.Suite
+	testContext *common.TestContext
+	configURL   string
+}
+
+func (suite *OSBCatalogTestSuite) SetupSuite() {
+	suite.testContext = common.NewTestContextBuilder().
+		WithEnvExtensions(func(e env.Environment, servers map[string]common.FakeServer) {
+			e.Set("director_gql.page_size", 3)
+		}).Build(suite.T())
+	suite.configURL = suite.testContext.Servers[common.DirectorServer].URL() + "/config"
+}
+
+func (suite *OSBCatalogTestSuite) SetupTest() {
+	http.DefaultClient.Post(suite.configURL+"/reset", "application/json", nil)
+}
+
+func (suite *OSBCatalogTestSuite) TearDownSuite() {
+	suite.testContext.CleanUp()
+}
+
+func (suite *OSBCatalogTestSuite) TestEmptyResponse() {
+	suite.configureResponse("query", "applications", appsEmptyResponse)
+	suite.testContext.SystemBroker.GET("/v2/catalog").WithHeader("X-Broker-API-Version", "2.15").Expect().
+		Status(http.StatusOK).
+		Body().Equal("{\"services\":[]}\n")
+}
+
+func (suite *OSBCatalogTestSuite) TestResponseWithOnePage() {
+	suite.configureResponse("query", "applications", appsMockResponse)
+	suite.testContext.SystemBroker.GET("/v2/catalog").WithHeader("X-Broker-API-Version", "2.15").Expect().
+		Status(http.StatusOK).
+		Body().Equal(appsExpectedCatalog)
+}
+
+func (suite *OSBCatalogTestSuite) TestResponseWithSeveralPages() {
+	suite.configureResponse("query", "applications", appsPageResponse1)
+	suite.configureResponse("query", "applications", appsPageResponse2)
+	suite.testContext.SystemBroker.GET("/v2/catalog").WithHeader("X-Broker-API-Version", "2.15").Expect().
+		Status(http.StatusOK).
+		Body().Equal(appsExpectedCatalogPaging)
+}
+
+func (suite *OSBCatalogTestSuite) TestErrorWhileFetchingApplicaitons() {
+	suite.configureResponse("query", "applications", appsPageResponse1)
+	suite.configureResponse("query", "applications", appsErrorResponse)
+
+	suite.testContext.SystemBroker.GET("/v2/catalog").WithHeader("X-Broker-API-Version", "2.15").Expect().
+		Status(http.StatusInternalServerError).
+		JSON().Object().Value("description").Equal("could not build catalog")
+}
+
+func (suite *OSBCatalogTestSuite) configureResponse(queryType, queryName, response string) {
+	var applicationsResponse map[string]interface{}
+
+	err := json.Unmarshal([]byte(response), &applicationsResponse)
+	assert.NoError(suite.T(), err)
+
+	body := common.ConfigRequestBody{
+		GraphqlQueryKey: common.GraphqlQueryKey{
+			Type: queryType,
+			Name: queryName,
+		},
+		Response: applicationsResponse,
+	}
+	jsonBody, err := json.Marshal(body)
+
+	assert.NoError(suite.T(), err)
+	http.DefaultClient.Post(suite.configURL, "application/json", bytes.NewReader(jsonBody))
```

Comment:
it will be probably better to not use the default client even in tests (it has no timeout, tests can hang, etc)